src/controller.cpp

```diff
@@ -184,6 +184,12 @@ bool Controller::activate() {
       m_portalDetected = false;
       return true;
     }
+
+    if (hasCooldownForAllServers()) {
```

Comment:
I think the root cause of the issue here is more to do with the fact that calling `Controller::disconnect()` doesn't necessarily lead to a disconnected signal if we fail before instructing the platform to connect. This doesn't seem like it entirely fixes the problem, and we may have similar edge cases in server switching where this could occur.